### PR TITLE
Add press release syndication to back to the hub site

### PR DESCRIPTION
**Pull request by pfrilling**
## Summary
This adds the functionality to send press releases back to the hub site using the Json API. It does this by:

1. Updating the `EcmsBase` class to allow file uploads
2. Adds a new `ecms_api_press_release_publisher` module that queue's press release nodes recursively on it's referenced entities (paragraphs/taxonomy terms).
3. It uses cron to create the referenced entities on the hub site, eventually creating the node once all dependencies have been created.
4. This requires the `paragraphs_type_permissions` module. Without this, creating paragraph entities through the Json API was not possible.
5. Update hooks were written to give paragraph permissions to the appropriate roles.
6. Unit tests were updated to cover the important changes (a bulk of the changes below).

## Metadata
| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | no
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | yes (mostly)
| Did you perform browser testing? | yes
| Risk level | High
| Relevant links | [RIG-81](https://thinkoomph.jira.com/browse/rig-81)

### CODE REVIEW

File: ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\ecms_api_press_release_publisher\PressReleasePublisher $pressReleasePublisher
+   *   The ecms_api_press_release_publisher service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, PressReleasePublisher $pressReleasePublisher) {
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php:56]: arg type for $plugin_id string

File: ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->pressReleasePublisher = $pressReleasePublisher;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function processItem($data): void {
```
Comment (ben-hamelin) [inline ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php:65]: Just double checking but pretty sure you can't add the type argument here due to the batch API right?

**Review by ben-hamelin — CHANGES_REQUESTED** [review]
Just two minor questions on typing, otherwise looking good!

File: ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php (hunk 2 above)
Comment (pfrilling) [reply ecms_base/modules/custom/ecms_api_press_release_publisher/src/Plugin/QueueWorker/PressReleaseQueueWorker.php:65]: PHPStorm isn't yelling at me for declaring the return type. My local cron testing was working too. I think I can leave it as is.

**Review by ben-hamelin — APPROVED** [review]
Ok looks good!
/approved

**Pull request merged**